tools/memap.py

```diff
@@ -361,56 +479,136 @@ def parse_map_file_iar(self, file_desc):
                 else:
                     self.module_add(name, size, section)
 
-    def search_objects(self, path):
-        """ Searches for object files and creates mapping: object --> module
+                if line.startswith('*** MODULE SUMMARY'): # finish section
+                    break
+
+            # Start decoding the map file
+            current_library = ""
+            for line in infile:
+
+                library = self.check_new_library_iar(line)
+
+                if library != "":
+                    current_library = library
+
+                object_name = self.check_new_object_lib_iar(line)
+
+                if object_name != "" and current_library != "":
+                    temp = '[lib]' + '/'+ current_library + '/'+ object_name
+                    self.module_replace(object_name, temp)
+
+
+    export_formats = ["json", "csv-ci", "table"]
+
+    def list_dir_obj(self, path):
+        """ Searches all objects in BUILD directory and creates list
 
         Positional arguments:
-        path - the path to an object file
+        path - the path to a map file
         """
 
         path = path.replace('\\', '/')
 
         # check location of map file
-        rex = r'^(.+)' + r'\/(.+\.map)$'
+        #rex = r'^(.+)' + r'\/(.+\.map)$'
+        rex = r'^(.+)\/(.+\.map)$'
```

Comment:
This comment is the same as the other Regular expression related ones:
1) `re` not `rex` is common in python
2) Please make this a Class or global constant